Core/Matcher/ContentMatcher.php

```diff
@@ -56,7 +56,7 @@ public function matchContent(array $conditions)
 
             // BC support
             if ($key == 'content_type') {
-                if (is_int($key) || ctype_digit($key)) {
+                if (ctype_digit($key)) {
```

Comment:
As we checked the $key 's value one line higher for being a string, this whole check is useless, no?